flink-connector/src/main/java/org/apache/gravitino/flink/connector/catalog/BaseCatalog.java

```diff
@@ -347,9 +353,9 @@ public List<String> listFunctions(String s) throws DatabaseNotExistException, Ca
   }
 
   @Override
-  public CatalogFunction getFunction(ObjectPath objectPath)
+  public CatalogFunction getFunction(ObjectPath tablePath)
       throws FunctionNotExistException, CatalogException {
-    throw new UnsupportedOperationException();
```

Comment:
`getFunction` `getXXStatistics` seems not related to this PR?

Reply:
getTableStatistics is needed.
```
FlinkHiveCatalogIT > testHivePartitionTable() FAILED
    org.apache.flink.table.api.ValidationException: SQL validation failed. null
        at org.apache.flink.table.planner.calcite.FlinkPlannerImpl.org$apache$flink$table$planner$calcite$FlinkPlannerImpl$$validate(FlinkPlannerImpl.scala:200)
        at org.apache.flink.table.planner.calcite.FlinkPlannerImpl.validate(FlinkPlannerImpl.scala:117)
        at org.apache.flink.table.planner.operations.SqlNodeToOperationConversion.convert(SqlNodeToOperationConversion.java:261)
        at org.apache.flink.table.planner.delegation.ParserImpl.parse(ParserImpl.java:106)
        at org.apache.flink.table.api.internal.TableEnvironmentImpl.executeSql(TableEnvironmentImpl.java:728)
        at org.apache.gravitino.flink.connector.integration.test.FlinkEnvIT.sql(FlinkEnvIT.java:159)
        at org.apache.gravitino.flink.connector.integration.test.hive.FlinkHiveCatalogIT.lambda$testHivePartitionTable$3(FlinkHiveCatalogIT.java:360)
        at org.apache.gravitino.flink.connector.integration.test.FlinkEnvIT.doWithSchema(FlinkEnvIT.java:175)
        at org.apache.gravitino.flink.connector.integration.test.hive.FlinkHiveCatalogIT.testHivePartitionTable(FlinkHiveCatalogIT.java:307)

        Caused by:
        java.lang.UnsupportedOperationException
            at org.apache.gravitino.flink.connector.catalog.BaseCatalog.getTableStatistics(BaseCatalog.java:387)
            at org.apache.flink.table.planner.catalog.DatabaseCalciteSchema.extractTableStats(DatabaseCalciteSchema.java:122)
            at org.apache.flink.table.planner.catalog.DatabaseCalciteSchema.getStatistic(DatabaseCalciteSchema.java:106)
            at org.apache.flink.table.planner.catalog.DatabaseCalciteSchema.lambda$getTable$0(DatabaseCalciteSchema.java:94)
            at java.util.Optional.map(Optional.java:215)
            at org.apache.flink.table.planner.catalog.DatabaseCalciteSchema.getTable(DatabaseCalciteSchema.java:90)
            at org.apache.calcite.jdbc.SimpleCalciteSchema.getImplicitTable(SimpleCalciteSchema.java:126)
            at org.apache.calcite.jdbc.CalciteSchema.getTable(CalciteSchema.java:295)
            at org.apache.calcite.sql.validate.EmptyScope.resolve_(EmptyScope.java:145)
            at org.apache.calcite.sql.validate.EmptyScope.resolveTable(EmptyScope.java:101)
            at org.apache.calcite.sql.validate.DelegatingScope.resolveTable(DelegatingScope.java:214)
            at org.apache.calcite.sql.validate.IdentifierNamespace.resolveImpl(IdentifierNamespace.java:116)
            at org.apache.calcite.sql.validate.IdentifierNamespace.validateImpl(IdentifierNamespace.java:188)
            at org.apache.calcite.sql.validate.AbstractNamespace.validate(AbstractNamespace.java:89)
            at org.apache.calcite.sql.validate.SqlValidatorImpl.validateNamespace(SqlValidatorImpl.java:1050)
            at org.apache.calcite.sql.validate.SqlValidatorImpl.validateQuery(SqlValidatorImpl.java:1025)
            at org.apache.calcite.sql.validate.SqlValidatorImpl.validateFrom(SqlValidatorImpl.java:3294)
            at org.apache.calcite.sql.validate.SqlValidatorImpl.validateFrom(SqlValidatorImpl.java:3276)
            at org.apache.calcite.sql.validate.SqlValidatorImpl.validateSelect(SqlValidatorImpl.java:3603)
            at org.apache.calcite.sql.validate.SelectNamespace.validateImpl(SelectNamespace.java:64)
            at org.apache.calcite.sql.validate.AbstractNamespace.validate(AbstractNamespace.java:89)
            at org.apache.calcite.sql.validate.SqlValidatorImpl.validateNamespace(SqlValidatorImpl.java:1050)
            at org.apache.calcite.sql.validate.SqlValidatorImpl.validateQuery(SqlValidatorImpl.java:1025)
            at org.apache.calcite.sql.SqlSelect.validate(SqlSelect.java:248)
            at org.apache.calcite.sql.validate.SqlValidatorImpl.validateScopedExpression(SqlValidatorImpl.java:1000)
            at org.apache.calcite.sql.validate.SqlValidatorImpl.validate(SqlValidatorImpl.java:749)
            at org.apache.flink.table.planner.calcite.FlinkPlannerImpl.org$apache$flink$table$planner$calcite$FlinkPlannerImpl$$validate(FlinkPlannerImpl.scala:196)
            ... 8 more

```

Reply:
got it